bot.py

```diff
@@ -230,6 +247,242 @@ async def resolveTownInfo(self, ctx, params):
 
         return (post, objs)
 
+
+    @commands.command(name='createTown', aliases=['createtown'], help='Create an entire town on this server, including categories, roles, channels, and permissions')
+    async def createTown(self, ctx):
+        params = shlex.split(ctx.message.content)
+
+        guild = ctx.guild
+        
+        usageStr = "Usage: `<town name> [server storyteller role] [server player role]`"
+
+        if len(params) < 2:
+            await ctx.send("Too few params to `!createTown`. " + usageStr)
+            return None
+
+        townName = params[1]        
+        if not townName:
+            await ctx.send("No town name provided. " + usageStr)
+            return None
+            
+        botRole = getRoleByName(guild, self.bot.user.name)
+        if not botRole:
+            await ctx.send("Could not find role for \"" + self.bot.user.name + "\". Cannot proceed! Where did the role go?")
+            return None
+        
+        serverStRole = len(params) > 2 and getRoleByName(guild, params[2]) or None
```

Comment:
why "server" on these variable names?

Reply:
These are not per-town, but are per-server roles, for like "A person who wants to play BotC" or "A person who wants to Storytell BotC".  They're optional, as some servers certainly are built just for playing BotC.

I guess I could have named them "guild"

Reply:
Yeah I’d prefer to use guild consistently since that’s the discord api lingo